container/debug.go

```diff
@@ -0,0 +1,258 @@
+package container
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"path/filepath"
+	"reflect"
+
+	"github.com/pkg/errors"
+
+	"github.com/goccy/go-graphviz"
+	"github.com/goccy/go-graphviz/cgraph"
+)
+
+// DebugOption is a functional debug option for a container.
+type DebugOption interface {
+	applyConfig(*debugConfig) error
```

Comment:
having a private method is to prevent other implementations, right?

Reply:
Yes

Reply:
maybe it's worth to expand the doc comment.

Reply:
I updated the doc comment a bit